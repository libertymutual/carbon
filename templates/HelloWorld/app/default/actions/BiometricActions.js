  import * as ActionTypes from '../actions/ActionTypes';

/**
 * Counter related action creators
 */
export default {

  /**
   * Updates the username value
   * @returns {function} A function which dispatches an UPDATE_USERNAME action
   */
  updateUsername(username) {
    return function (dispatch) {
      dispatch({ type: ActionTypes.UPDATE_USERNAME, username });
    };
  },

  /**
   * Updates the password value
   * @returns {function} A function which dispatches an UPDATE_PASSWORD action
   */
  updatePassword(password) {
    return function (dispatch) {
      dispatch({ type: ActionTypes.UPDATE_PASSWORD, password });
    };
  },

  /**
   * Updates state to reflect that credentials have been saved
   * @returns {function} A function which dispatches a SAVE_CREDENTIALS action
   */
  saveCredentials() {
    return function (dispatch) {
      dispatch({ type: ActionTypes.SAVE_CREDENTIALS });
    };
  },

  /**
   * Updates state to reflect that credentials have been saved
   * @returns {function} A function which dispatches a DELETE_CREDENTIALS action
   */
  deleteCredentials() {
    return function (dispatch) {
      dispatch({ type: ActionTypes.DELETE_CREDENTIALS });
    };
  },
};
