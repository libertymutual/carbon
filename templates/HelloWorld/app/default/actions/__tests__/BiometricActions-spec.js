import 'react-native';
import React from 'react';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import BiometricActions from '../BiometricActions';

describe('BiometricActions tests', () => {

  const middlewares = [thunk];
  const mockStore = configureMockStore(middlewares);
  let store;
  let dispatch;

  beforeEach(() => {
    // Sets up a new store
    store = mockStore();
    dispatch = store.dispatch;
  });

  test('updateUsername returns a function', () => {
    var dispatcher = BiometricActions.updateUsername();
    expect(typeof dispatcher).toMatchSnapshot();
  });

  test('updateUsername function dispatches an UPDATE_USERNAME action', () => {
    var dispatcher = BiometricActions.updateUsername();
    dispatch(dispatcher);
    expect(store.getActions()).toMatchSnapshot();
  });

  test('updatePassword returns a function', () => {
    var dispatcher = BiometricActions.updatePassword();
    expect(typeof dispatcher).toMatchSnapshot();
  });

  test('updatePassword function dispatches an UPDATE_PASSWORD action', () => {
    var dispatcher = BiometricActions.updatePassword();
    dispatch(dispatcher);
    expect(store.getActions()).toMatchSnapshot();
  });

  test('saveCredentials returns a function', () => {
    var dispatcher = BiometricActions.saveCredentials();
    expect(typeof dispatcher).toMatchSnapshot();
  });

  test('saveCredentials function dispatches an SAVE_CREDENTIALS action', () => {
    var dispatcher = BiometricActions.saveCredentials();
    dispatch(dispatcher);
    expect(store.getActions()).toMatchSnapshot();
  });

  test('deleteCredentials returns a function', () => {
    var dispatcher = BiometricActions.deleteCredentials();
    expect(typeof dispatcher).toMatchSnapshot();
  });

  test('deleteCredentials function dispatches an DELETE_CREDENTIALS action', () => {
    var dispatcher = BiometricActions.deleteCredentials();
    dispatch(dispatcher);
    expect(store.getActions()).toMatchSnapshot();
  });

});
