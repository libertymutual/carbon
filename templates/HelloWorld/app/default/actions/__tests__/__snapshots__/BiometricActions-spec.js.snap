// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BiometricActions tests deleteCredentials function dispatches an DELETE_CREDENTIALS action 1`] = `
Array [
  Object {
    "type": "delete_credentials",
  },
]
`;

exports[`BiometricActions tests deleteCredentials returns a function 1`] = `"function"`;

exports[`BiometricActions tests saveCredentials function dispatches an SAVE_CREDENTIALS action 1`] = `
Array [
  Object {
    "type": "save_credentials",
  },
]
`;

exports[`BiometricActions tests saveCredentials returns a function 1`] = `"function"`;

exports[`BiometricActions tests updatePassword function dispatches an UPDATE_PASSWORD action 1`] = `
Array [
  Object {
    "password": undefined,
    "type": "update_password",
  },
]
`;

exports[`BiometricActions tests updatePassword returns a function 1`] = `"function"`;

exports[`BiometricActions tests updateUsername function dispatches an UPDATE_USERNAME action 1`] = `
Array [
  Object {
    "type": "update_username",
    "username": undefined,
  },
]
`;

exports[`BiometricActions tests updateUsername returns a function 1`] = `"function"`;
